instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -107,7 +119,58 @@ class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::cla
         val countDownLatch = CountDownLatch(1)
         runOnMainSync {
             mapboxNavigation.historyRecorder.stopRecording {
-                logE("history path=$it", "DEBUG")
+                logE("history path=$it", LOG_CATEGORY)
+                countDownLatch.countDown()
+            }
+        }
+        countDownLatch.await()
+
+        // Verify alternative routes events were triggered.
+        assertEquals(2, routes.size)
+        assertTrue(mapboxNavigation.getNavigationRoutes().isNotEmpty())
+        firstAlternative.verifyOnRouteAlternativesPassed()
+        firstAlternative.alternatives!!.forEach {
+            assertFalse(routes.contains(it))
+        }
+    }
+
+    @Test
+    fun alternative_routes_observer_is_not_triggered_for_set_routes() {
```

Comment:
I noticed that you use deprecated `setRoutes`/`getRoutes`. I think it would be a good idea to use the actual API in new tests. Old ones can be left as is while the old API works but why increase its usage?

Reply:
it's needed to respect `RouteAlternativesIdlingResource` which handles `DirectionsRoute`s. I would change the whole flow to `NavigationRoutes`, let's do it separately

Reply:
Okay